platform/cms/accelerated_publishing/rfc/reverse-proxy-falback.md

```diff
@@ -0,0 +1,93 @@
+
+We then added files to the "new" bucket, such that we would expect increasing amounts of hits with respect to the new bucket, and therefore anticipated faster response times. Our testing confirmed that eliminating the cycle of 404 response and request to the fallback location did improve response times. The average difference between a 100% miss rate (slower) and a 100% hit rate (faster) was **0.07 seconds** total.
+
+
+
+## Risks
+1. **Latency.** As discussed in the Proof of concept section, our primary concern was that this design obviously introduces latency to any request that we expect to be filled by the Content Build S3 bucket, since a 404 request/response cycle from the Next Build S3 bucket is required in order for traffic to subsequently reach the Content Build bucket.
+
+    As described above, we ran tests on an equivalent configuration using two S3 buckets, and found that the average latency introduced to the Content Build (i.e. fallback) requests was **0.07 seconds**. This amount of latency is non-zero but does not represent a significant change in Veteran experience.
+
+    We'd also note that this latency is only introduced for Content Build content. As content is moved to Next Build, it will effectively be delivered faster by no longer being subject to the fallback case.
+
+2. **Stability** Obviously, any change to a critical system infrastructure like the reverse proxy comes with risks and the potential for unforeseen consequence. It is our hope that this can be mitigated and addressed by evaluation by Platform Devops COP members who are most familiar with the reverse proxy and larger traffic flow concerns. Any recommendations would then fall to the Accelerated Publish team to implement.
+
+## Alternatives
+**1. Load balancing with rules**. Prior to coming to the nginx fallback solution, we explored the idea of using a load balancer to direct traffic to one S3 bucket or another. In theory, load balancer rules could be used to send traffic to one place or another.
+
+There were two primary issues with this approach:
+
+1.  The number of load balancer rules required would be beyond what AWS ALB's handle. There is not a clean correspondance between the URLs/routes of VA.gov content and the type of content it is. It is not possible to clearly state 'I want all content of type X to go to one place' based on URL.
+
+    Given that, there would need to be an individual rule for each URL to determine which bucket that request would go to. Next Build is already building over 10,000 pages of Content Build content. This would require a number of ALB rules that AWS does not support. AWS ALB's support 100 rules by default. This is adjustable, but in our case it would need to be adjusted multiple times to accommodate content growth.
```

Comment:
I don't know if it's worth expanding here at all, but as an example there are 141 VAMC facilities. Each have multiple content types underneath their umbrella. The number of rules gets out of hand quickly even if it isn't 1 route: 1 rule.